lib/enumerable_method.rb

```diff
@@ -0,0 +1,200 @@
+# This is our custom fake enumerables
+module Enumerable
+  # my_each
+  def my_each(&block)
+    return enum_for(:my_each) unless block_given?
+
+    for item in self
+      block.call item
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |data|
+      yield data, i
+
+      i += 1
+    end
+    self
+  end
+
+  # my_select function
+  def my_select
```

Comment:
On the #my_select method, we only got the following error:
```
rspec ./spec/enumerables_spec.rb:123 # enumerables #my_select returns an array containing all elements of enum for which the given block returns a true value::range

```

Reply:
fixed for instance of Range